lib/Crypt/SysRandom.pm

```diff
@@ -17,6 +17,7 @@ if (eval { require Crypt::SysRandom::XS }) {
 	sub random_bytes_win32 {
 		my ($count) = @_;
 		return '' if $count == 0;
+		Carp::croak('Could not read more than 256 bytes of randomness') if $count > 256;
```

Comment:
This needs to be documented, with a brief explanation about the limit. (I assume a similar change should be made to the XS module.)

I don't think you need to worry about putting the amount requested in the error message, especially since you are using `croak`. If it's the same, I'd be tempted to use a constant.